flink-clients/src/test/java/org/apache/flink/client/deployment/application/FromJarEntryClassInformationProviderTest.java

```diff
@@ -72,9 +73,13 @@ public void testEitherJobClassNameOrJarHasToBeSet() {
         FromJarEntryClassInformationProvider.createFromCustomJar(null, null);
     }
 
-    @Ignore // FLINK_OPT_DIR has to be set
     @Test
     public void testPythonJarFile() {
+        // TODO: see FLINK-23154
+        Assume.assumeTrue(
+                "This tests only succeeds if " + ConfigConstants.ENV_FLINK_OPT_DIR + " is set.",
+                System.getProperty(ConfigConstants.ENV_FLINK_OPT_DIR) != null);
```

Comment:
Maybe we need to use `System.getenv`. Refer to `PackagedProgramUtils#getPythonJar`.

Reply:
Yeah, I thought of building a classpath that includes a mocked `flink-python.jar`. But adapting the enviroment variables is not advisable. That's were I stopped investigating.